public/modules/custom/helfi_rekry_content/helfi_rekry_content.module

```diff
@@ -120,24 +122,67 @@ function _helfi_rekry_content_get_media_image(string|NULL $fid = NULL): ?string
 }
 
 /**
- * Get media entity by file id.
+ * Validate and return video url, used in migration.
  *
  * @param string|null $url
  *   The video url.
  *
  * @return string|null
  *   Valid video url or null
+ *
+ * @throws \Drupal\migrate\MigrateSkipRowException
  */
 function _helfi_rekry_content_get_video_url(string|NULL $url = NULL): ?string {
   try {
     $resolver = \Drupal::service('media.oembed.url_resolver');
-    $validate = $resolver->getProviderByUrl($url);
-    return $url;
+    /** @var \Drupal\media\OEmbed\Provider $provider */
+    $provider = $resolver->getProviderByUrl($url);
   }
   catch (\throwable $e) {
     \Drupal::logger('helfi_rekry_content')->notice('Video embed url "' . $url . '" failed validation with message: ' . $e->getMessage());
     throw new MigrateSkipRowException();
   }
+
+  // Ticket #UHF-9069 prevent migrating bad oembed links.
+  try {
+    // Use the same validation used in field validation.
+    $resource_url = $resolver->getResourceUrl($url);
+    $resource = \Drupal::service('media.oembed.resource_fetcher')
+      ->fetchResource($resource_url);
+    return $url;
+  }
+  catch (ResourceException | ProviderException $e) {
+    // fetchResource fails, the link is no good.
+    if (!str_contains(strtolower($provider->getName()), 'youtube')) {
+      \Drupal::logger('helfi_rekry_content')
+        ->error('Bad video url rejected by oembed-validation: ' . $url);
+      throw new MigrateSkipRowException();
+    }
+
+    // Remove all extra query parameters from youtube link.
+    $querystring = parse_url($url, PHP_URL_QUERY);
+    $queryParameters = [];
+    parse_str($querystring, $queryParameters);
+    if ($video_id = $queryParameters['v']) {
+      $parts = parse_url($url);
+      $path = $parts['path'];
+      $host = $parts['host'];
+      $scheme = $parts['scheme'] ? "{$parts['scheme']}://" : '';
+      $query = http_build_query([
+        'v' => $video_id,
+      ]);
+      $new_url = sprintf('%s%s%s?%s', $scheme, $host, rtrim($path, '/'), $query);
```

Comment:
This gives warnings when url does not contain scheme. Then `parse_url` interprets domain as part of the path.

```php
$parts = parse_url('www.youtube.com/watch?app=desktop&v=lN3Uv8xmAXc');
$host = $parts['host']; // =>  WARNING  Undefined array key "host" in ...
$scheme = $parts['scheme']; // =>  WARNING  Undefined array key "scheme" in ...
```

Also, the resulting $new_url is invalid. Media library does not accept it without scheme:

<img width="809" alt="image" src="https://github.com/City-of-Helsinki/drupal-helfi-rekry/assets/732227/813c433d-e50e-49fa-ab01-0cf229235ebf">

I propose we add `https://` scheme to the url if it does not have scheme:

```php
if (strpos($url, "://") === FALSE) {
  $url = "https://$url";
}
```